Neos.ContentRepository.Core/Classes/SharedModel/Privilege/ContentStreamPrivilege.php

```diff
@@ -0,0 +1,35 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Neos\ContentRepository\Core\SharedModel\Privilege;
+
+use Neos\ContentRepository\Core\SharedModel\Workspace\ContentStreamIds;
+
+/**
+ * @internal except for custom PrivilegeProviderInterface implementations
+ */
+final class ContentStreamPrivilege
+{
+    private function __construct(
+        public readonly ?ContentStreamIds $allowedContentStreamIds,
+        public readonly ?ContentStreamIds $disallowedContentStreamIds,
```

Comment:
what is the semantics of having both of these lists? :) I don't fully understand this yet.

Reply:
Me neither :)
But I thought, that might need both. an _allow_ list and a _deny_ list..

Reply:
Looking at this after a break, I think the terms "allowed" and "disallowed" are not helpful.
i.e. what does `$privileges->isContentStreamAllowed($contentStreamId)` exactly mean?

Also we should re-evaluate whether we really need allow- _and_ deny list